src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -86,12 +85,59 @@ public ModelAndView classSetup(@PathVariable String sectionId,
         return page;
     }
 
+    private ModelAndView validateInputType(String sectionId, CourseConfigurationForm classSetupForm) {
+        if (classSetupForm.getGradingOn()){
+            boolean typeValid = (classSetupForm.getAssignmentPoints().matches("[0-9.]+") || classSetupForm.getAssignmentPoints() == null)
+                    && (classSetupForm.getPresentPoints().matches("[0-9.]+") || classSetupForm.getPresentPoints() == null)
+                    && (classSetupForm.getExcusedPoints().matches("[0-9.]+") || classSetupForm.getExcusedPoints() == null)
+                    && (classSetupForm.getAbsentPoints().matches("[0-9.]+") || classSetupForm.getAbsentPoints() == null)
+                    && (classSetupForm.getTardyPoints().matches("[0-9.]+") || classSetupForm.getTardyPoints() == null);
+
+            if(!typeValid){
+                if(!classSetupForm.getAssignmentPoints().matches("[0-9.]+") || classSetupForm.getAssignmentPoints() != null) {
+                    ModelAndView page = new ModelAndView("/courseConfiguration");
+                    page.addObject("error", "The Total Points field contained an incorrect value. Please enter a valid number.");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "error" 9 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)